src/ai/defaultai_warfare.cc

```diff
@@ -985,30 +985,95 @@ bool DefaultAI::check_militarysites(const Time& gametime) {
 	   militarysites.front().built_time + Duration(10 * 60 * 1000) < gametime &&
 	   bf.military_loneliness < 1000 - 10 * std::abs(management_data.get_military_number_at(14));
 
-	bool should_be_dismantled = false;
-	const int32_t enemy_military_capacity = std::max<int32_t>(
-	   {bf.enemy_military_presence,
-	    bf.enemy_military_sites * (1 + std::abs(management_data.get_military_number_at(77) / 20)),
-	    (bf.enemy_owned_land_nearby) != 0u ?
-          4 + std::abs(management_data.get_military_number_at(99) / 20) :
-          0});
-	if (bf.enemy_owned_land_nearby != 0u) {
-		if (bf.military_score_ < std::abs(management_data.get_military_number_at(91) * 10) &&
-		    bf.area_military_capacity - static_cast<int16_t>(total_capacity) -
-		          std::abs(management_data.get_military_number_at(84) / 10) >
-		       (std::abs(management_data.get_military_number_at(24) / 25) + 1) *
-		          enemy_military_capacity) {
-			should_be_dismantled = true;
-		}
-	} else {
-		const uint16_t size_bonus =
-		   total_capacity * std::abs(management_data.get_military_number_at(89)) / 5;
-		if (bf.military_score_ + size_bonus < management_data.get_military_number_at(88) * 5 &&
-		    bf.area_military_capacity > static_cast<int16_t>(total_capacity)) {
-			should_be_dismantled = true;
-		}
+	// inputs for decisions
+	// bf.enemy_owned_land_nearby - decreases dismantle probability
+	// bf.unowned_land_nearby - decreases dismantle probability
+    // portspace_nearby - decreases dismantle probability
+	
+	// bf.future_area_military_capacity - increases dismantle probability
+	// bf.military_loneliness - increases dismantle probability
+	// bf.future_military_loneliness - increases dismantle probability
+	// soldier_status_ 
+	// bf.military_score_ - decreases dismantle probability
+	// bf.own_military_presence - increases dismantle probability
+	
+	// bf.enemy_military_presence - decreases dismantle probability
+	// bf.enemy_military_sites - - decreases dismantle probability
+
+	// available military numbers: 77, 99, 91, 84, 24, 89, 88
+	// available neuron pools: 13,14, 28,29, 30, 45 - 48, 57 - 59
+
+	// We calculate the score - if positive, site will be dismounted
+	// We starting with positive number, most modificators are decreasing th score
+	int32_t dismantle_score = 3 * management_data.get_military_number_at(91) / 2 + 100;
+	dismantle_score -= 2 * management_data.neuron_pool[13].get_result_safe(
+		                        bf.enemy_owned_land_nearby / 2, kAbsValue);
+	dismantle_score -= (bf.enemy_owned_land_nearby) ?
```

Comment:
the bool expression needs to be expressed explicitly as bool.
i.e. bf.enemy_owned_land_nearby > 0 
otherwise clang-tidy should complain

Reply:
got it